### PR TITLE
Serialize queries execution over osqueryd socket

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Serializes queries execution over osqueryd socket.

Addresses the issue: https://github.com/elastic/beats/issues/25297

In this particular case the two schedules queries were executed at the same time on the same socket that caused the problems such as:

```
osquery failed: write unix ->/var/run/642722736/osquery.sock: write: broken pipe
```
```
: *osquery.ExtensionResponse error reading struct: error reading field 0: read unix ->/var/run/404419649/osquery.sock: i/o timeout
```

Currently the queries are scheduled by osquerybeat. Going forwards the handling of the scheduled queries will move into osqueryd for 7.14.

## Why is it important?

Fixed the problem in 7.13 release as described in https://github.com/elastic/beats/issues/25297

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas

## How to test this PR locally
- Steps to Reproduce: 
1. Add osquery manager integration to the policy
2. Enroll and run the agent
3. Add two scheduled queries
Screen Shot 2021-04-25 at 10 25 03 AM

4. Restart the agent 
5. Check osquerybeat log. Before the fix you would see failures on scheduled queries execution, ad-hoc query would fail too. After this fix the queries should be executed as expected no errors.

## Related issues

- Closes https://github.com/elastic/beats/issues/25297

### CODE REVIEW

**Review by james-elastic — APPROVED** [review]

File: x-pack/osquerybeat/beater/osquerybeat.go [hunk 1]
```diff
@@ -49,6 +51,9 @@ type osquerybeat struct {
 	// Beat lifecycle context, cancelled on Stop
 	cancel context.CancelFunc
 	mx     sync.Mutex
+
+	// limiter to run one query at a time
+	queryLimiterCh chan struct{}
```
Comment (urso) [inline x-pack/osquerybeat/beater/osquerybeat.go:56]: The way it is initialized and used looks like a counting semaphore. 

I think Heartbeat uses this type: https://pkg.go.dev/golang.org/x/sync/semaphore

Comment (aleksmaus) [reply x-pack/osquerybeat/beater/osquerybeat.go:56]: Checked the implementation of that, it's more than I need and didn't like this property:
"If ctx is already done, Acquire may still succeed without blocking."
https://pkg.go.dev/golang.org/x/sync/semaphore#Weighted.Acquire

Comment (urso) [reply x-pack/osquerybeat/beater/osquerybeat.go:56]: This property is only true if all these conditions are met:
- the current go-routine is waiting on acquire
- The ctx is closed at the same time as acquire succeeds
- Another go-routine did actually call release when the context was closed.

As the `select` clause randomized checking order, your code actually has the same property, but worse. The Acquire fails if the ctx was closed before. Your select based approach can succeed even if the context was already closed.

Comment (aleksmaus) [reply x-pack/osquerybeat/beater/osquerybeat.go:56]: Yep, the query below will fail with cancelled context. Will update to sem.

**Review by scunningham — APPROVED** [review]
Lookin' good in the neighborhood.

**Review by urso — APPROVED** [review]

**Pull request merged**